src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java

```diff
@@ -165,8 +165,12 @@ private void runEventLoop(SelectedSelectionKeySet keys)
       int size = tasks.size();
       try {
         if (size > 0) {
+          Runnable task;
           while (alive && size > 0) {
```

Comment:
Yes, the check is needed, if there are no tasks, we call `selector.select()` that will block, otherwise, we call `selector.selectNow()` that returns immediately.